### ISSUE TITLE
All transformers are called even if target is specified

### ISSUE THREAD

**Issue by jogelin** [BioPhoton/angular-formatter-parser#7]
Hello,

I specified multiple transformers:
 ```
  fPConfig: IFormatterParserConfig = {
    formatterParser: [
      { name: 'dateToString', target: 0  },
      { name: 'stringToDate', target: 1  }
    ]
  };
```

And I though that if I correctly specified the targets:
* 0 : If the model is modified => call of the `dateToString` to format the view
* 1: If the view is modify => call of the `stringToDate` to format the model

But both are called all the time: dateToString => stringToDate

And stringToDate is not called with the value from dateToString but with the same value.

See example (check console)
https://stackblitz.com/edit/angular-kum1wv 

Is it normal ?
I would like to have one formatter for : model => view
and one parser for: view => model

**Comment by BioPhoton** [comment]
That's normal. 
If you type into an input box you change the view value. 
When a controls view value is changed, the forms model value get's updated with the new value.

You can provide transform functions for the view and model values that hook into the flow described above and transform the value before they get assigned to eighter the view or the model.

Your example with slight changes here:
[https://stackblitz.com/edit/angular-dcxeqh](https://stackblitz.com/edit/angular-dcxeqh)

**Comment by BioPhoton** [comment]
If you have any other questions feel free to ask or reopen the issue.

best,
Michael

**Comment by jogelin** [comment]
We can keep it closed but I still misunderstanding the flow ;)

I agree, if the model change => call dateToString
Then the view has changes so => call stringToDate

But the stringToDate should be called with the transformed value of the view no (from dateToString) ? And not the value of the model...

To have a better example, in my case, I would like to have a date in my model and a formatted date on my view, for me the cycle should be 

Model => dateToString(date: Date) => '01/01/2015' in the input => stringToDate(date: string) => Date in the model

But I have something like:

If the model change:
dateToString(date: **Date**) => '01/01/2015' in the input
THEN
stringToDate(date: **Date**) => crash because the date is not a string from the view but a real date from the model


If the view change:
stringToDate(date: **string**) => crash because the date is not the one of the model but a simple string
THEN
dateToString(date: **string**) => real Date in the model

Below an example if you want to play:
https://stackblitz.com/edit/angular-cmz3od

I would like to use your very good library in parallel with formly and create a custom dynamic component field. I can put an example if I could fix that ;)

**Comment by jogelin** [comment]
@BioPhoton Any suggestions ? :s

**Comment by Tukks** [comment]
@jogelin it's because directive listen for input with @HostListener, et the only thing you get in that way in the event throw by the input is what the user type.
in pseudo code : 

> @HostListener('input', ['$event'])
>   onControlInput($event: KeyboardEvent) {
>     inputElement.setValue = transformValue(inputElement.value)
>    model.setValue = transformValue(inputElement.value)
> }

I think @BioPhoton do that because if you read the model value, you can't know where the user put the new value.

**Comment by jogelin** [comment]
@Tukks Ok I understand thx. 

So the target option is always 2, both ways ? If yes, it is a bit useless true ?

**Comment by Tukks** [comment]
No, for example if you put thousand separator
Your config will be : 

> this.numberConfig = {
>       formatterParser: [
>         // 0 for view, 1 for model, 2 or nothing for both
>         { name: 'addSeparatorInteger', target: 0},
>         { name: 'removeSeparatorInteger', target: 1}
>       ]
>     };

the first function get the value from input, transform it and put in the value.
the second one get the value from the input, transform it and put in the model.
example : 
1) user input 1234 ---> transformView = 1 124, transformModel = 1124
2) user input a another number like 0, you receive in your 2 different method as argument : "1 1240"
so -->  transformView = 112 40, transformModel = 11240

In conclusion : if you put target 0, the method will be applied only to transformView, 1 is for transformModel and 2 for both.

try it, doing custom transform method, and watch in debugger what you reiceve as parameter.

I hope i'm clear.

**Comment by jogelin** [comment]
@Tukks It's clear ! Thx for your help ;)

**Comment by BioPhoton** [comment]
Hi @Tukks big thanks for your help in clarifying the inner principles of this lib!
I see you read at least the core code of this lib. :-)

best Michael
